.github/workflows/push-pypi.yml

```diff
@@ -0,0 +1,57 @@
+name: Publish to Test PyPI.org
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - master
```

Comment:
tag tigger maybe better?

Reply:
@yihong0618 can you kindly clarify? we want this to only release to PyPI on a release and when you make the release you will make a tag. Otherwise it pushes to test PyPI when something is merged to master to test the build and deploy steps.

Reply:
like 
```yml
name: Release

on:
  push:
    tags:
      - "*"
```
when we add new tags, we publish new version to PyPI

Reply:
oh gotcha - in this case @yihong0618 we want to push to PYPI when we make a release. Essentially what will happen is we create a release and when we do that it will have a new tag value so we will create a new tag there. this will force us to follow a proper release workflow where a tagged release also has changelog notes.

Also when i link up zenodo to the package a release will also trigger an updated DOI associated with the changes. so in this case we do want this to be on release not on tag.